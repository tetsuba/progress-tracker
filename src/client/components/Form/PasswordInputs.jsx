import React from 'react'
import { Form } from 'react-bootstrap';

// COMPONENTS
import PasswordStrength from '../PasswordStrength/PasswordStrength';

// UTILS
import { passwordMatchError } from './form-utils'

export default function PasswordInputs(props) {
  const { register, inputs, setInputs } = props
  const passwordMissMatchError = passwordMatchError(inputs)

  return (
    <>
      <Form.Group controlId="newPassword" className="mb-3">
        <Form.Label className="text-muted">
          { register ? 'Password': 'New password'}
        </Form.Label>
        <Form.Control
          required
          type="password"
          placeholder="******"
          name="newPassword"
          onChange={setInputs}
          value={inputs.newPassword}
        />
        <PasswordStrength password={inputs.newPassword} />
      </Form.Group>
      <Form.Group controlId="confirmPassword" className="mb-3">
        <Form.Label className="text-muted">Confirm password</Form.Label>
        <Form.Control
          required
          type="password"
          placeholder="******"
          name="confirmPassword"
          onChange={setInputs}
          value={inputs.confirmPassword}
          isInvalid={passwordMissMatchError}
        />
        <Form.Control.Feedback type="invalid">
          {passwordMissMatchError && 'Passwords do not match!!!'}
        </Form.Control.Feedback>
        <Form.Text className="text-muted">
          {`Matching password ${inputs.confirmPassword.length} of ${inputs.newPassword.length}`}
        </Form.Text>
      </Form.Group>
    </>
  )
}