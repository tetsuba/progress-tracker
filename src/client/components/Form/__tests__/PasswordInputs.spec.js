import { testRenderer } from '../../../../test/testHelper'
import PasswordInputs from '../PasswordInputs'

describe('<PasswordInputs>', () => {
  const baseProps = {
    setInputs: jest.fn(),
    inputs: {
      newPassword: '',
      confirmPassword: '',
    },
  }

  describe('@Render', () => {
    it('should render', () => {
      const props = {
        ...baseProps,
      }
      const wrapper = testRenderer(PasswordInputs, props)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
