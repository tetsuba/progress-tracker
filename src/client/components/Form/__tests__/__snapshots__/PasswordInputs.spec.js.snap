// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordInputs> @Render should render 1`] = `
<Fragment>
  <FormGroup
    className="mb-3"
    controlId="newPassword"
  >
    <FormLabel
      className="text-muted"
      column={false}
      srOnly={false}
    >
      New password
    </FormLabel>
    <FormControl
      name="newPassword"
      onChange={[MockFunction]}
      placeholder="******"
      required={true}
      type="password"
      value=""
    />
    <PasswordStrength
      password=""
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
    controlId="confirmPassword"
  >
    <FormLabel
      className="text-muted"
      column={false}
      srOnly={false}
    >
      Confirm password
    </FormLabel>
    <FormControl
      isInvalid={false}
      name="confirmPassword"
      onChange={[MockFunction]}
      placeholder="******"
      required={true}
      type="password"
      value=""
    />
    <Feedback
      type="invalid"
    />
    <FormText
      className="text-muted"
    >
      Matching password 0 of 0
    </FormText>
  </FormGroup>
</Fragment>
`;
